Habitica/src/main/java/com/habitrpg/android/habitica/ui/activities/FullProfileActivity.java

```diff
@@ -424,27 +424,38 @@ public void gotGear(List<ItemData> obj, HabitRPGUser user) {
 
             // Calculate class bonus
             String itemClass = i.getKlass();
+            String itemSpecialClass = i.getSpecialClass();
 
-            if (itemClass.isEmpty()) {
+            if (itemClass.isEmpty() && itemSpecialClass.isEmpty()) {
                 continue;
             }
 
+            float classBonus = 0.5f;
+            Boolean userClassMatchesGearClass = itemClass.equals(user.getClass());
+            Boolean userClassMatchesGearSpecialClass = itemSpecialClass.equals(user.getClass());
+
+            if (!userClassMatchesGearClass && !userClassMatchesGearSpecialClass) classBonus = 0;
+
```

Comment:
I think that now, by the time the code reaches this point, classBonus always has the right value for handling this particular gear piece (i.e., if the gear piece belongs to the rogue class and the player is a rogue, then classBonus is 0.5; for any other combination, classBonus is 0). That means that the switch statement is no longer needed. All you need is:
```
	conClassBonus = con_ * classBonus;
	intClassBonus = int_ * classBonus;
	perClassBonus = per_ * classBonus;
	strClassBonus = str_ * classBonus;
```
It doesn't matter that that code is calculating bonuses for attributes that aren't relevant for the user's class because those attributes are going to be set to 0 and so the class bonus for them will end up as 0.